{{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/users/tests/test_urls.py

```diff
@@ -7,10 +7,7 @@
 
 
 def test_detail(user: User):
-    assert (
-        reverse("users:detail", kwargs={"username": user.username})
-        == f"/users/{user.username}/"
-    )
+    assert user.get_absolute_url() == f"/users/{user.username}/"
```

Comment:
Not sure about this... The spirit of this test is to make sure the URL reverses properly. Now we're introducing a dependency on `get_absolute_url()`

Reply:
Yes, I understand that but the `detail view` would always use the `get_absolute_url()` method throughout the project to get the detail view of the concerned object so wouldn't it make more sense to directly invoke that method and test it? What if someone adds a `userPassesTest` Mixin etc to not allow everyone to get the detail view? In that scenario, one would either need to update this test or add another test for this.

Reply:
How about this one? It's not using `get_absolute_url()` is it?

https://github.com/pydanny/cookiecutter-django/blob/9ebfd5102c008debaf8721f90262a6d741c26898/%7B%7Bcookiecutter.project_slug%7D%7D/%7B%7Bcookiecutter.project_slug%7D%7D/templates/base.html#L81

> What if someone adds a userPassesTest Mixin etc to not allow everyone to get the detail view? In that scenario, one would either need to update this test or add another test for this.

I'm not sure I understand what you mean. Requesting the page is a different thing than reversing the URL. I think this type of mixin is added to the view. If not satisfied, it prevents view access, it doesn't prevent reversing the URL location.

Again, I think this is testing detail _URL route_, not the detail _view_.

Reply:
I see what you mean now. I was definitely testing the view with this change than the URL. I will revert the change in `test_urls.py`
Regarding the line, I would change it to:
`<a class="nav-link" href="request.user.get_absolute_url">{% trans "My Profile" %}</a>`

because I think one should reduce as many moving parts as possible. I know doing this might be a bit unconventional as I have also never seen anyone use `get_absolute_url()` to get the `detail-view-url` as opposed to using the usual URL template tag and supplying the pk and the URL name. 

Out of curiosity, is there any potential issue I am missing? Because it seems to me that with this change, the template code would be a bit cleaner, way more stable especially if the `slug_field` were ever to change or some permission mixin was added to the `get_absolute_url()` method etc. Those cases do seem not super likely though so that could be a reason for this.

Reply:
Regarding the template tag vs `get_absolute_url` in the template... I really don't mind either way. We use the `url` tag several other times in this file, seems consistent as it is. 

If things were to change, one would have to find usages and update them, yes, but I don't see that as a compelling reason to change the template.

Reply:
Ok makes sense. Thank you for your reply.